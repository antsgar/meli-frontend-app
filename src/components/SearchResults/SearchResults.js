import { Breadcrumb, ResultsCard, SpinnerSupport } from "..";
import styles from "./SearchResults.module.scss";

const SearchResults = ({ categories, items }) => {
    return (
        <div className={styles.container}>
            <div className={styles.content}>
                <SpinnerSupport>
                    <Breadcrumb breadcrumbItems={categories} />
                    <ResultsCard results={items} />
                </SpinnerSupport>
            </div>
        </div>
    );
};

export default SearchResults;
