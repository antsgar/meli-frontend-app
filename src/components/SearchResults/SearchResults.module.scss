@import "../../styles/mixins";
@import "../../styles/vars";

.container {
    @include gridWrapper(12);

    background-color: $background-color-default;
    padding-bottom: $spacing-large;
    flex: 1;

    @include desktop {
        padding-left: $spacing-large;
        padding-right: $spacing-large;
    }
}

.content {
    @include gridContent(2, 12);
    margin-bottom: $spacing-large;
}
