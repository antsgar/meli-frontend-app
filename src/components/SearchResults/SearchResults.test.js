import React from "react";
import { mount } from "enzyme";
import SearchResults from "./SearchResults";
import { categories, items } from "../../fixtures";

let wrapper;

beforeEach(() => {
    wrapper = mount(<SearchResults items={items} categories={categories} />);
});

describe("SearchResults", () => {
    it("should render component", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should display product categories", () => {
        const breadcrumb = wrapper.find("Breadcrumb");
        expect(breadcrumb.exists()).toEqual(true);
        expect(breadcrumb.find(".item")).toHaveLength(categories.length);
    });

    it("should display results card", () => {
        expect(wrapper.find("ResultsCard").exists()).toEqual(true);
    });
});
