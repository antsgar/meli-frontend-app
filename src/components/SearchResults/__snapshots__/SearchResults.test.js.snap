// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults should render component 1`] = `
<SearchResults
  categories={
    Array [
      "Celulares y Teléfonos",
      "Celulares y Smartphones",
    ]
  }
  items={
    Array [
      Object {
        "condition": "new",
        "free_shipping": true,
        "id": "MLA860381954",
        "picture": "http://mla-s1-p.mlstatic.com/885508-MLA32445302787_102019-I.jpg",
        "price": Object {
          "amount": 166799,
          "currency": "ARS",
          "decimals": null,
        },
        "state_name": "Capital Federal",
        "title": "iPhone 11 64 Gb Negro 4 Gb Ram",
      },
      Object {
        "condition": "new",
        "free_shipping": true,
        "id": "MLA859045066",
        "picture": "http://mla-s1-p.mlstatic.com/639283-MLA32646972173_102019-I.jpg",
        "price": Object {
          "amount": 288900,
          "currency": "ARS",
          "decimals": null,
        },
        "state_name": "Capital Federal",
        "title": "iPhone 11 Pro Max 256 Gb Gris Espacial 4 Gb Ram",
      },
      Object {
        "condition": "new",
        "free_shipping": true,
        "id": "MLA846526085",
        "picture": "http://mla-s2-p.mlstatic.com/662389-MLA31003000984_062019-I.jpg",
        "price": Object {
          "amount": 149989.99,
          "currency": "ARS",
          "decimals": 99,
        },
        "state_name": "Capital Federal",
        "title": "iPhone 8 Plus 64 Gb Gris Espacial 3 Gb Ram",
      },
      Object {
        "condition": "new",
        "free_shipping": true,
        "id": "MLA860817466",
        "picture": "http://mla-s1-p.mlstatic.com/633990-MLA31002678135_062019-I.jpg",
        "price": Object {
          "amount": 158000,
          "currency": "ARS",
          "decimals": null,
        },
        "state_name": "Capital Federal",
        "title": "iPhone XR 64 Gb Negro 3 Gb Ram",
      },
    ]
  }
>
  <div
    className="container"
  >
    <div
      className="content"
    >
      <SpinnerSupport>
        <div
          className="container"
        >
          <Breadcrumb
            breadcrumbItems={
              Array [
                "Celulares y Teléfonos",
                "Celulares y Smartphones",
              ]
            }
          >
            <div
              className="container"
            >
              <span
                className="item"
                key="Celulares y Teléfonos"
              >
                Celulares y Teléfonos
              </span>
              <span
                className="item"
                key="Celulares y Smartphones"
              >
                Celulares y Smartphones
              </span>
            </div>
          </Breadcrumb>
          <ResultsCard
            results={
              Array [
                Object {
                  "condition": "new",
                  "free_shipping": true,
                  "id": "MLA860381954",
                  "picture": "http://mla-s1-p.mlstatic.com/885508-MLA32445302787_102019-I.jpg",
                  "price": Object {
                    "amount": 166799,
                    "currency": "ARS",
                    "decimals": null,
                  },
                  "state_name": "Capital Federal",
                  "title": "iPhone 11 64 Gb Negro 4 Gb Ram",
                },
                Object {
                  "condition": "new",
                  "free_shipping": true,
                  "id": "MLA859045066",
                  "picture": "http://mla-s1-p.mlstatic.com/639283-MLA32646972173_102019-I.jpg",
                  "price": Object {
                    "amount": 288900,
                    "currency": "ARS",
                    "decimals": null,
                  },
                  "state_name": "Capital Federal",
                  "title": "iPhone 11 Pro Max 256 Gb Gris Espacial 4 Gb Ram",
                },
                Object {
                  "condition": "new",
                  "free_shipping": true,
                  "id": "MLA846526085",
                  "picture": "http://mla-s2-p.mlstatic.com/662389-MLA31003000984_062019-I.jpg",
                  "price": Object {
                    "amount": 149989.99,
                    "currency": "ARS",
                    "decimals": 99,
                  },
                  "state_name": "Capital Federal",
                  "title": "iPhone 8 Plus 64 Gb Gris Espacial 3 Gb Ram",
                },
                Object {
                  "condition": "new",
                  "free_shipping": true,
                  "id": "MLA860817466",
                  "picture": "http://mla-s1-p.mlstatic.com/633990-MLA31002678135_062019-I.jpg",
                  "price": Object {
                    "amount": 158000,
                    "currency": "ARS",
                    "decimals": null,
                  },
                  "state_name": "Capital Federal",
                  "title": "iPhone XR 64 Gb Negro 3 Gb Ram",
                },
              ]
            }
          >
            <div>
              <Result
                key="MLA860381954"
                result={
                  Object {
                    "condition": "new",
                    "free_shipping": true,
                    "id": "MLA860381954",
                    "picture": "http://mla-s1-p.mlstatic.com/885508-MLA32445302787_102019-I.jpg",
                    "price": Object {
                      "amount": 166799,
                      "currency": "ARS",
                      "decimals": null,
                    },
                    "state_name": "Capital Federal",
                    "title": "iPhone 11 64 Gb Negro 4 Gb Ram",
                  }
                }
              >
                <div
                  className="hoverable"
                  data-id="MLA860381954"
                  data-test="result-item"
                  onClick={[Function]}
                >
                  <div
                    className="container"
                  >
                    <div
                      className="content"
                    >
                      <img
                        alt="Product picture"
                        className="picture"
                        src="http://mla-s1-p.mlstatic.com/885508-MLA32445302787_102019-I.jpg"
                      />
                      <div
                        className="productInfoContainer"
                      >
                        <div
                          className="priceAndStateContainer"
                        >
                          <div
                            className="priceAndShipping"
                          >
                            ARS 166,799
                            <img
                              alt="Free shipping icon"
                              className="freeShippingIcon"
                              src="./ic_shipping@2x.png.png"
                            />
                          </div>
                          <div
                            className="stateName"
                          >
                            Capital Federal
                          </div>
                        </div>
                        <div
                          className="title"
                        >
                          iPhone 11 64 Gb Negro 4 Gb Ram
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </Result>
              <Result
                key="MLA859045066"
                result={
                  Object {
                    "condition": "new",
                    "free_shipping": true,
                    "id": "MLA859045066",
                    "picture": "http://mla-s1-p.mlstatic.com/639283-MLA32646972173_102019-I.jpg",
                    "price": Object {
                      "amount": 288900,
                      "currency": "ARS",
                      "decimals": null,
                    },
                    "state_name": "Capital Federal",
                    "title": "iPhone 11 Pro Max 256 Gb Gris Espacial 4 Gb Ram",
                  }
                }
              >
                <div
                  className="hoverable"
                  data-id="MLA859045066"
                  data-test="result-item"
                  onClick={[Function]}
                >
                  <div
                    className="container"
                  >
                    <div
                      className="content"
                    >
                      <img
                        alt="Product picture"
                        className="picture"
                        src="http://mla-s1-p.mlstatic.com/639283-MLA32646972173_102019-I.jpg"
                      />
                      <div
                        className="productInfoContainer"
                      >
                        <div
                          className="priceAndStateContainer"
                        >
                          <div
                            className="priceAndShipping"
                          >
                            ARS 288,900
                            <img
                              alt="Free shipping icon"
                              className="freeShippingIcon"
                              src="./ic_shipping@2x.png.png"
                            />
                          </div>
                          <div
                            className="stateName"
                          >
                            Capital Federal
                          </div>
                        </div>
                        <div
                          className="title"
                        >
                          iPhone 11 Pro Max 256 Gb Gris Espacial 4 Gb Ram
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </Result>
              <Result
                key="MLA846526085"
                result={
                  Object {
                    "condition": "new",
                    "free_shipping": true,
                    "id": "MLA846526085",
                    "picture": "http://mla-s2-p.mlstatic.com/662389-MLA31003000984_062019-I.jpg",
                    "price": Object {
                      "amount": 149989.99,
                      "currency": "ARS",
                      "decimals": 99,
                    },
                    "state_name": "Capital Federal",
                    "title": "iPhone 8 Plus 64 Gb Gris Espacial 3 Gb Ram",
                  }
                }
              >
                <div
                  className="hoverable"
                  data-id="MLA846526085"
                  data-test="result-item"
                  onClick={[Function]}
                >
                  <div
                    className="container"
                  >
                    <div
                      className="content"
                    >
                      <img
                        alt="Product picture"
                        className="picture"
                        src="http://mla-s2-p.mlstatic.com/662389-MLA31003000984_062019-I.jpg"
                      />
                      <div
                        className="productInfoContainer"
                      >
                        <div
                          className="priceAndStateContainer"
                        >
                          <div
                            className="priceAndShipping"
                          >
                            ARS 149,990
                            <img
                              alt="Free shipping icon"
                              className="freeShippingIcon"
                              src="./ic_shipping@2x.png.png"
                            />
                          </div>
                          <div
                            className="stateName"
                          >
                            Capital Federal
                          </div>
                        </div>
                        <div
                          className="title"
                        >
                          iPhone 8 Plus 64 Gb Gris Espacial 3 Gb Ram
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </Result>
              <Result
                key="MLA860817466"
                result={
                  Object {
                    "condition": "new",
                    "free_shipping": true,
                    "id": "MLA860817466",
                    "picture": "http://mla-s1-p.mlstatic.com/633990-MLA31002678135_062019-I.jpg",
                    "price": Object {
                      "amount": 158000,
                      "currency": "ARS",
                      "decimals": null,
                    },
                    "state_name": "Capital Federal",
                    "title": "iPhone XR 64 Gb Negro 3 Gb Ram",
                  }
                }
              >
                <div
                  className="hoverable"
                  data-id="MLA860817466"
                  data-test="result-item"
                  onClick={[Function]}
                >
                  <div
                    className="container"
                  >
                    <div
                      className="content"
                    >
                      <img
                        alt="Product picture"
                        className="picture"
                        src="http://mla-s1-p.mlstatic.com/633990-MLA31002678135_062019-I.jpg"
                      />
                      <div
                        className="productInfoContainer"
                      >
                        <div
                          className="priceAndStateContainer"
                        >
                          <div
                            className="priceAndShipping"
                          >
                            ARS 158,000
                            <img
                              alt="Free shipping icon"
                              className="freeShippingIcon"
                              src="./ic_shipping@2x.png.png"
                            />
                          </div>
                          <div
                            className="stateName"
                          >
                            Capital Federal
                          </div>
                        </div>
                        <div
                          className="title"
                        >
                          iPhone XR 64 Gb Negro 3 Gb Ram
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </Result>
            </div>
          </ResultsCard>
        </div>
      </SpinnerSupport>
    </div>
  </div>
</SearchResults>
`;
